scripts/haddocs.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+# Build haddock documentation and an index for all projects in
+# `ouroboros-network` repository.
+#
+# usage:
+# ./haddocks.sh directory [true|false]
+#
+# $1 - where to put the generated pages, this directory contents will be wiped
+#      out (so don't pass `/` or `./` - the latter will delete your 'dist-newstyle')
+#      (the default is './haddocks')
+# $2 - weahter to re-build haddocjs with `cabal haddock` command or a component name
+#      (the default is true)
+
+set -euo pipefail
+
+OUTPUT_DIR=${1:-"./haddocks"}
+REGENERATE=${2:-"true"}
+
+GHC_VERSION=$(ghc --numeric-version)
+
+if [ ${OUTPUT_DIR} == "/" ]; then
+  echo "Oh no, you don't want to run 'rm -rf /*'"
+  exit 1
+else
+  mkdir -p ${OUTPUT_DIR}
+  rm -rf ${OUTPUT_DIR}/*
+fi
```

Comment:
This looks so scary.
I only found a single place where `scripts/haddocs.sh` is called and at the call-site `${OUTPUT_DIR}` is generated from scratch anyway.
Maybe just add a test that `${OUTPUT_DIR}` is empty here and exit the script if that precondition is not met.

Reply:
Right, this was actually meant to work slightly differently. I'll fix this - though this still will be a bit scary.

Reply:
> Maybe just add a test that ${OUTPUT_DIR} is empty here and exit the script if that precondition is not met.

We don't need to, we create a directory with `mkdir` - it will fail if it already exists.

I wanted to remove the directory to avoid stale files, but we can just avoid doing this; The worse that can happen is that there will have some `.html` pages that are not linked from anywhere.